packages/plugins/rpc-interceptors/src/lib/eth_accounts.ts

```diff
@@ -0,0 +1,72 @@
+import { Account, Embark } from "embark-core";
+const { blockchain: blockchainConstants } = require("embark-core/constants");
+import RpcInterceptor from "./rpcInterceptor";
+import { ProxyResponseParams } from "embark-proxy";
+import { AccountParser, dappPath } from "embark-utils";
+import Web3 from "web3";
+
+export default class EthAccounts extends RpcInterceptor {
+
+  public _accounts: Account[] | null = null;
+
+  public _nodeAccounts: string[] | null = null;
+
+  private _web3: Web3 | null = null;
+
+  constructor(embark: Embark) {
+    super(embark);
+  }
+
+  protected get web3() {
+    return (async () => {
+      if (!this._web3) {
+        await this.events.request2("blockchain:started");
+        // get connection directly to the node
+        const provider = await this.events.request2("blockchain:node:provider", "ethereum");
+        this._web3 = new Web3(provider);
+      }
+      return this._web3;
+    })();
+  }
+
+  private get accounts() {
+    return (async () => {
+      if (!this._accounts) {
+        const web3 = await this.web3;
+        const nodeAccounts = await this.nodeAccounts;
+        this._accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, nodeAccounts);
+      }
+      return this._accounts || [];
+    })();
+  }
+
+  private get nodeAccounts() {
+    return (async () => {
+      if (!this._nodeAccounts) {
+        const web3 = await this.web3;
+        this._nodeAccounts = await web3.eth.getAccounts();
+      }
+      return this._nodeAccounts || [];
+    })();
+  }
```

Comment:
You'll see `accounts, `nodeAccounts` and `web3` accounts being implemented in many interceptors. @emizzle originally those were injected via `MutationOptions` but was then decided against. What was the reasoning again? I remember @iurimatias proposed that this wasn't a good idea and you eventually agreed.

Wonder if we might want to revisit the injection...

Reply:
Not sure why the injection was removed, but I see we also have the `resetAccount` function only in RpcModifier. I think it means tha ton config reset and account reset (like in the tests), most of these modifiers/intereceptors won't work.

Reply:
For the classes **that extend `RpcInterceptor`**, `web3`, `accounts`, and `nodeAccounts` should live in the base `RpcInterceptor` class. This is for convenience when intercepting all of the core Embark RPC requests/responses. Additionally, it serves other functions such as updating the account list when node accounts are created via `personal_newAccount`, and when accounts are reset during tests (as @jrainville explained above).

For other plugins that want to intercept an RPC request/response that **do not extend `RpcInterceptor`**, they will need to define their own `web3`, `accounts`, and `nodeAccounts` as needed. For these cases, I originally tried to get these injected via `MutationOptions`, but it wasn't a good idea.